### PR TITLE
feat(bitmap): store bitmap in meta store

### PR DESCRIPTION
**Pull request by soundOfDestiny**
## What's changed and what's your intention?

store bitmap in meta store

## Checklist

- [x] I have written necessary docs and comments

### CODE REVIEW

**Review by zwang28 — COMMENTED** [review]
LGTM. 
Can we also inspect the maximum size of a `HummockVersion `, e.g. during a e2e test, in case it's already too big for this change.

**Comment by soundOfDestiny** [comment]
> LGTM.
> Can we also inspect the maximum size of a `HummockVersion `, e.g. during a e2e test, in case it's already too big for this change.

where?

**Comment by zwang28** [comment]
> > LGTM.
> > Can we also inspect the maximum size of a `HummockVersion `, e.g. during a e2e test, in case it's already too big for this change.
> 
> where?

May need to add metrics and also increase `SST_BITMAP_LIMIT` I think.

**Comment by soundOfDestiny** [comment]
> > > LGTM.
> > > Can we also inspect the maximum size of a `HummockVersion `, e.g. during a e2e test, in case it's already too big for this change.
> > 
> > 
> > where?
> 
> May need to add metrics and also increase `SST_BITMAP_LIMIT` I think.

where can we calculate and report size of `HummockVersion `?

File: src/storage/src/hummock/sstable/builder.rs [hunk 1]
```diff
@@ -165,7 +177,22 @@ impl SSTableBuilder {
             version: VERSION,
         };
 
-        (self.buf.freeze(), meta)
+        (
+            self.buf.freeze(),
+            meta,
+            if self.vnode_bitmaps.len() > SST_BITMAP_LIMIT {
+                vec![]
```
Comment (hzxa21) [inline src/storage/src/hummock/sstable/builder.rs:184]: `SST_BITMAP_LIMIT` is 0 and we will also reach here to produce empty bitmap

**Comment by hzxa21** [comment]
> > > > LGTM.
> > > > Can we also inspect the maximum size of a `HummockVersion `, e.g. during a e2e test, in case it's already too big for this change.
> > > 
> > > 
> > > where?
> > 
> > 
> > May need to add metrics and also increase `SST_BITMAP_LIMIT` I think.
> 
> where can we calculate and report size of `HummockVersion `?

I think we can create a metric for `HummockVersion` size and update the metric in `trigger_commit_stat`, which will be triggered on `add_tables` and `commit_epoch`.

File: src/storage/src/hummock/sstable/builder.rs (hunk 1 above)
Comment (soundOfDestiny) [reply src/storage/src/hummock/sstable/builder.rs:184]: fixed. now `SST_BITMAP_LIMIT` == 10

File: src/storage/src/hummock/shared_buffer/shared_buffer_uploader.rs [hunk 2]
```diff
@@ -169,14 +169,15 @@ impl SharedBufferUploader {
...
-            .map(|sst| SstableInfo {
+            .map(|(sst, vnode_bitmap)| SstableInfo {
                 id: sst.id,
                 key_range: Some(risingwave_pb::hummock::KeyRange {
                     left: sst.meta.smallest_key.clone(),
                     right: sst.meta.largest_key.clone(),
                     inf: false,
                 }),
                 file_size: sst.meta.estimated_size as u64,
+                vnode_bitmap: vnode_bitmap.clone(),
```
Comment (hzxa21) [inline src/storage/src/hummock/shared_buffer/shared_buffer_uploader.rs:180]: can we use into_iter and avoid the clone?

**Review by hzxa21 — APPROVED** [review]
Rest LGTM

File: src/storage/src/hummock/shared_buffer/shared_buffer_uploader.rs (hunk 2 above)
Comment (soundOfDestiny) [reply src/storage/src/hummock/shared_buffer/shared_buffer_uploader.rs:180]: thanks. #2230

**Pull request merged**